src/Microsoft.Azure.ServiceBus/Constants.cs

```diff
@@ -25,16 +23,12 @@ static class Constants
 
         public const int DefaultClientPrefetchCount = 0;
 
-        public static readonly int MaximumMessageHeaderPropertySize = ushort.MaxValue;
```

Comment:
Unfortunately, this we do this check only in service side right now. We do not get the serialized value size for the header from the AMQP library. We pass the amqp header objects to the library to do the serialization which gets serialized and we get the final serialized message. So we are able to enforce only the max message size on the client. This message will end up going to server, and server side validation will be done for these.